src/richie/apps/courses/models/course.py

```diff
@@ -398,6 +398,51 @@ def get_course_runs_for_language(self, language=None):
 
         return course_runs
 
+    def get_course_runs_state(self):
+        """
+        Returns a dict of course runs with their state.
+        """
+        run_open = {}
+        run_upcoming = {}
+        run_ongoing = {}
+        run_archived = {}
+        run_unknown = {}
+        now = timezone.now()
+        course_runs = self.get_course_runs_for_language()
+        for run in course_runs:
```

Comment:
As discussed during our meetup today, maybe something like:
```python
course_runs_dict = {i: [] for i in range(6)}
for run in course_runs:
    course_runs_dict[run.state["priority"]].append(run)
```

Reply:
I applied your solution to not recalcute the state. The "priority" was a value I did not understand you could retrieve from coursestate. Thanks for pointing it out. Adjusted the course_detail html in consequence. The newest commit should be good. I also rebased it.